doc/source/user_guide/categorical.rst

```diff
@@ -263,6 +263,20 @@ All instances of ``CategoricalDtype`` compare equal to the string ``'category'``
 
    c1 == "category"
 
+Notice that the ``categories_dtype`` should be considered, especially when comparing with
+two empty ``CategoricalDtype`` instances.
+
+.. ipython:: python
+
+    import numpy as np
+
+    c2 = pd.Categorical(np.array([], dtype=object))
+    c3 = pd.Categorical(np.array([], dtype=float))
+
+    # c2.dtype = CategoricalDtype(categories=[], ordered=False, categories_dtype=object)
+    # c3.dtype = CategoricalDtype(categories=[], ordered=False, categories_dtype=float64)
```

Comment:
I don't think these are necessary, but if you'd prefer to have them then can you instead just do e.g. `c2.dtype` on a line of its own and have the docs build produce the output.